### ISSUE TITLE
Use list markup for the listbox contents

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#22890]
## Open questions

- The dropdowns are styled really differently. 
  - Is it as simple as using some of the same classes to get consistent with VADS combobox? 
  - Or at least consistent within this form? 
  - Or is that a bigger lift that should be ticketed separately?

---
## Description
The Location, VA Health > Service type, and Community provider > service type inputs currently don’t use list markup for the contents of the dropdown. VoiceOver still announces how many list items are in the dropdown without list markup, though other screen readers may not (and may not even if we adopt more semantic HTML, but I am hoping it [fixes this odd ‘1 of 1’ announcement](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22830#issuecomment-3603513929)). Using list markup is more semantic and therefore likely to be more [robust](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Guides/Understanding_WCAG/Robust). 

- **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 4
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User encounters components and patterns that meet best practices

## To Reproduce

- Go to [staging facility locator prototype](https://staging.va.gov/find-locations/)
- Check out the HTML used in each custom dropdown.
  - Note that the location and VA Health > Service type inputs use a wrapping div as the listbox and paragraphs for the options
  - Note that the Community provider > service type uses divs for the listbox and the options.

## Screenshots

VADS combox dropdown using list markup:
<img width="1163" height="325" alt="Image" src="https://github.com/user-attachments/assets/91c418f4-36d1-447e-82d7-0f48fb49f3cb" />

<details>
<summary>Screenshots of current dropdown markup</summary>

Location dropdown HTML:
<img width="466" height="368" alt="Image" src="https://github.com/user-attachments/assets/fe4fa419-a307-4105-9825-f41aa1189f92" />

VA Health > Service type:
<img width="1155" height="434" alt="Image" src="https://github.com/user-attachments/assets/cc42263e-ef1d-424a-9615-5d14ff846217" />

Community provider > service type dropdown:
<img width="1151" height="406" alt="Image" src="https://github.com/user-attachments/assets/6d963fbc-e7ff-4a65-8afb-70724c12b82b" />

</details>

## Recommendation
Convert those dropdowns to use `<ul>` for the listbox and `<li>` for the options.

## AC / Expected behavior

- [ ] Location dropdown uses list markup 
- [ ] VA Health > Service type dropdown uses list markup
- [ ] Community provider > service type dropdown uses list markup

Bonus: 
- [ ] Confirm if the [odd ‘1 of 1’ announcement using Narrator](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22830#issuecomment-3603513929)) persists.
  - [ ] If it does, create a ticket to look into it, but don't hold this ticket up. 

## Definition of done
- [ ] Code completed; PR created
- [ ] Review instance created/available; internal team notified to start reviews
- [ ] Code review (meets established quality and security standards)
- [ ] Design review (meets design & UX requirements)
- [ ] A11y review (meets accessibility standards)
- [ ] Functional/feature review (meets acceptance criteria)
- [ ] Automated tests passed; PR approved
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by r-bartlett-gsa** [comment]
@carlyMhayter With RI issues resolved, moving this back to In Progress. Please let me know once it's ready for QA.
